core/src/main/scala/anorm/Flattener.scala

```diff
@@ -0,0 +1,559 @@
+package anorm
+
+sealed trait Flattener[T] {
```

Comment:
Generalized from former `TupleFlattener` to be able to flatten in any case, whatever is the number of column (including a single one).
Generalisation useful for function application in any case.